micro-rdk/src/common/ota.rs

```diff
@@ -414,84 +428,98 @@ impl<S: OtaMetadataStorage> OtaService<S> {
         let mut got_info = false;
 
         log::info!("writing new firmware to address `{:#x}`", self.address,);
+        let mut stream = response.into_data_stream();
+
+        loop {
+            match stream
+                .try_next()
+                .map_err(FrameError::Network)
+                .or(async {
+                    async_io::Timer::after(Duration::from_secs(FRAME_TIMEOUT_SECS)).await;
+                    Err(FrameError::Timeout(FRAME_TIMEOUT_SECS as usize))
```

Comment:
the goal is to timeout if a single frame is unresponsive for more than 30seconds, unblocking the caller (ConfigMonitor) so it can try again completely in 10 seconds (ConfigMonitor's default period)